src/samples/audio/smart_amp_test_ipc4.c

```diff
@@ -37,57 +53,48 @@ typedef void (*smart_amp_proc)(int8_t const *src_ptr,
 struct smart_amp_data {
 	struct sof_smart_amp_ipc4_config ipc4_cfg;
 	struct sof_smart_amp_config config;
-	struct comp_data_blob_handler *model_handler;
-	void *data_blob;
-	size_t data_blob_size;
 	smart_amp_proc process;
 	uint32_t out_channels;
 };
 
+#ifdef __MODULE_BUILD__
+static struct smart_amp_data smart_amp_priv;
+#endif
+
 static int smart_amp_init(struct processing_module *mod)
 {
 	struct smart_amp_data *sad;
-	struct comp_dev *dev = mod->dev;
 	struct module_data *mod_data = &mod->priv;
-	const size_t in_size = sizeof(struct ipc4_input_pin_format) * SMART_AMP_NUM_IN_PINS;
-	const size_t out_size = sizeof(struct ipc4_output_pin_format) * SMART_AMP_NUM_OUT_PINS;
 	int ret;
 	const struct ipc4_base_module_extended_cfg *base_cfg = mod_data->cfg.init_data;
 
-	comp_dbg(dev, "smart_amp_init()");
```

Comment:
lets keep all logging in the built-in code otherwise its more difficult to integrate/debug. i.e. this one can be move below to line 72.